kotlinx-coroutines-core/common/src/flow/internal/DiscardingCollector.kt

```diff
@@ -0,0 +1,26 @@
+package kotlinx.coroutines.flow.internal
+
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.flow.FlowCollector
+
+/**
+ * A [FlowCollector] specifically built to be used for `Flow.dropWhileBusy`
+ *
+ * This collector ensures that the fist element is always sent. It then offers subsequent elements,
+ * allowing them to be dropped.
+ */
+@InternalCoroutinesApi
+internal class DiscardingCollector<T>(private val channel: SendChannel<T>) : FlowCollector<T> {
+
+    private var hasSentFirstValue: Boolean = false
+
+    override suspend fun emit(value: T) {
```

Comment:
The idea looks good to me. It seems very appropriate for the way collection from the flow conceptually works, where you write `flow.collect { value -> someCode(value) }` and `someCode` is called (and can become busy) only when the first element is there. However, it can be surprising for cases like `flow.onStart { delay(1000) }.collect { ... }` where it would still wait for the actual collection to start and would not consider the flow to be "busy" for this initial 1000 ms. So, it should be explicitly mentioned in the docs of `dropWhileBusy`, that is give there a more detailed explanation of what "busy" means for this operator.